app/screens/home/channel_list/categories_list/categories/body/category_body.tsx

```diff
@@ -25,25 +24,20 @@ type Props = {
 
 const extractKey = (item: ChannelModel) => item.id;
 
-const CategoryBody = ({sortedChannels, category, hiddenChannelIds, limit, onChannelSwitch, unreadChannels}: Props) => {
+const CategoryBody = ({sortedChannels, category, limit, onChannelSwitch, unreadChannels}: Props) => {
     const serverUrl = useServerUrl();
     const ids = useMemo(() => {
-        let filteredChannels = sortedChannels;
-
-        // Remove all closed gm/dms
-        if (hiddenChannelIds.size) {
-            filteredChannels = sortedChannels.filter((item) => item && !hiddenChannelIds.has(item.id));
-        }
+        const filteredChannels = sortedChannels;
 
         if (category.type === DMS_CATEGORY && limit > 0) {
             return filteredChannels.slice(0, limit);
         }
         return filteredChannels;
-    }, [category.type, limit, hiddenChannelIds, sortedChannels.length]);
+    }, [category.type, limit, sortedChannels]);
 
     const directChannels = useMemo(() => {
         return ids.concat(unreadChannels).filter(isDMorGM);
-    }, [ids.length, unreadChannels.length]);
+    }, [ids, unreadChannels]);
```

Comment:
I do understand the previous reasoning but that should not apply for this one nor the other one